### PR TITLE
Include Select Query to TCK [2/5]

### PR DESCRIPTION
**Pull request by otaviojava**
This pull request significantly expands the `FruitRepository` interface and its test coverage in the Jakarta Data TCK. It introduces new repository methods for more advanced querying (including support for `IN`, projections, ordering, and use of the `id` function), and adds comprehensive tests to validate these capabilities, including handling of edge cases and unsupported operations.

**Repository enhancements:**

* Added new query methods to `FruitRepository` for:
  - Finding fruits by a set of names using the `IN` clause with a `Set<String>` parameter.
  - Counting all fruits.
  - Projecting only the `name` attribute, ordered by quantity.
  - Projecting both `name` and `quantity` attributes, ordered by name.
  - Using the `id` function for lookup and ordering.
  - Filtering by name and by quantity with ordering.

**Test coverage improvements:**

* Added tests in `JakartaQueryTests` for:
  - The new `IN` query using a parameter collection, with fallback for unsupported databases.
  - Counting all fruits and verifying the result.
  - Using the `id` function in queries and ordering.
  - Projecting only name, and name+quantity, with correct ordering and result mapping.
  - Ensuring queries return empty results when no entities match.
  - Combining filtering and ordering in queries.
  - Added a private record `FruitTuple` for convenient tuple mapping in projection tests.

**Imports and setup:**

* Updated imports to support new features (`Set`, `Optional`). [[1]](diffhunk://#diff-3b54bdb1a7c563cbb7b203290749d3d33ca4c58bb2e4b01391b3e5c120c72ac4R23-R25) [[2]](diffhunk://#diff-6543dd8da246f8ec31bb4b5054b9ddbecf75c405f4aef41a763253982791a327R34)

### CODE REVIEW

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java [hunk 1]
```diff
@@ -322,6 +323,32 @@ void shouldIn() {
         }
     }
 
+    @DisplayName("should test in")
+    @Assertion(id = "1318",
+            strategy = "Execute an IN comparison on the name attribute with multiple values, this time using Set to " +
+                    "multiple parameters, " +
+                    "asserting membership in the provided set or accepting UnsupportedOperationException.")
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java]: Some of the language did not make sense. Rewriting as:
```suggestion
            strategy = "Execute a query that performs an IN comparison of the name " +
                    "attribute against a Set of multiple values, asserting membership " +
                    "in the provided set or accepting UnsupportedOperationException.")
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java [hunk 2]
```diff
@@ -371,4 +398,165 @@ void shouldOr() {
             }
         }
     }
+
+    @DisplayName("should test count")
+    @Assertion(id = "1318",
+            strategy = "Execute a count all Fruits" +
+                    "asserting every returned entity matches at least one predicate or accepting UnsupportedOperationException.")
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java]: This didn't make sense. There aren't any returned entities. The return value is a count of entities.
```suggestion
            strategy = "Execute a query that counts all entities in the database, " +
                    "asserting the count matches the total number of entities " +
                    "inserted or accepting UnsupportedOperationException.")
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java [hunk 3]
```diff
@@ -371,4 +398,165 @@ void shouldOr() {
...
+            } else {
+                throw exp;
+            }
+        }
+    }
+
+    @DisplayName("should count all Fruits using id function")
+    @Assertion(id = "1318",
+            strategy = "Execute a query by the id attribute using the id function, once it is an id attribute all " +
+                    "databases should support it")
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java]: Corrected some wording
```suggestion
            strategy = "Execute a query that performs an equality comparison on the "
                    "id attribute using the id function. Because it is an id attribute, all " +
                    "databases should support it")
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java [hunk 4]
```diff
@@ -322,6 +323,32 @@ void shouldIn() {
         }
     }
 
+    @DisplayName("should test in")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java]:
```suggestion
    @Assertion(id = "458",
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java [hunk 5]
```diff
@@ -371,4 +398,165 @@ void shouldOr() {
             }
         }
     }
+
+    @DisplayName("should test count")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java]:
```suggestion
    @Assertion(id = "458",
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java [hunk 6]
```diff
@@ -371,4 +398,165 @@ void shouldOr() {
...
+            if (type.isKeywordSupportAtOrBelow(DatabaseType.COLUMN)) {
+                // Column and Key-Value databases might not be capable of count.
+            } else {
+                throw exp;
+            }
+        }
+    }
+
+    @DisplayName("should count all Fruits using id function")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java:425]:
```suggestion
    @Assertion(id = "458",
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java [hunk 7]
```diff
@@ -371,4 +398,165 @@ void shouldOr() {
...
+                    "databases should support it")
+    void shouldFindByIdUsingIdFunction() {
+        var fruit = fruits.getFirst();
+        var result = fruitRepository.findByIdUsingIdFunction(fruit.getId());
+
+        Assertions.assertThat(result).isNotEmpty().get().isEqualTo(fruit);
+    }
+
+    @DisplayName("should return only name attribute order by quantity")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java]:
```suggestion
    @Assertion(id = "458",
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java [hunk 8]
```diff
@@ -371,4 +398,165 @@ void shouldOr() {
...
+            if (type.isKeywordSupportAtOrBelow(DatabaseType.COLUMN)) {
+                // Column and Key-Value databases might not be capable of sorting.
+            } else {
+                throw exp;
+            }
+        }
+    }
+
+    @DisplayName("should return only name and quantity attributes order by quantity")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java]:
```suggestion
    @Assertion(id = "458",
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java [hunk 9]
```diff
@@ -371,4 +398,165 @@ void shouldOr() {
...
+            if (type.isKeywordSupportAtOrBelow(DatabaseType.COLUMN)) {
+                // Column and Key-Value databases might not be capable of sorting.
+            } else {
+                throw exp;
+            }
+        }
+    }
+
+    @DisplayName("should return id using id function order by id")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java]:
```suggestion
    @Assertion(id = "458",
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java [hunk 10]
```diff
@@ -371,4 +398,165 @@ void shouldOr() {
...
+            if (type.isKeywordSupportAtOrBelow(DatabaseType.COLUMN)) {
+                // Column and Key-Value databases might not be capable of sorting.
+            } else {
+                throw exp;
+            }
+        }
+    }
+
+    @DisplayName("should return empty result when condition does not match any entity")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java]:
```suggestion
    @Assertion(id = "458",
```

File: tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java [hunk 11]
```diff
@@ -371,4 +398,165 @@ void shouldOr() {
...
+            if (type.isKeywordSupportAtOrBelow(DatabaseType.COLUMN)) {
+                // Column and Key-Value databases might not be capable querying by non-key attributes.
+            } else {
+                throw exp;
+            }
+        }
+    }
+
+    @DisplayName("should apply where condition and order by together")
+    @Assertion(id = "1318",
```
Comment (njr-11) [inline tck/src/main/java/ee/jakarta/tck/data/standalone/entity/JakartaQueryTests.java]:
```suggestion
    @Assertion(id = "458",
```

**Comment by otaviojava** [comment]
@njr-11 is this one ok now to go?

If yes, could you approve it, please?

We have more 3 that I would like to release on M2.
